apps/src/music/views/MiniMusicPlayer.tsx

```diff
@@ -81,25 +89,55 @@ const MiniPlayerView: React.FunctionComponent<MiniPlayerViewProps> = ({
 
     // Play sounds
     playerRef.current.playSong(sequencerRef.current.getPlaybackEvents());
+    setCurrentProjectId(project.id);
+  }, []);
+
+  const onStopSong = useCallback(async () => {
+    playerRef.current.stopSong();
+    setCurrentProjectId(undefined);
   }, []);
 
   // Some loading UI while we're fetching the library
   if (isLoading) {
     return <div>Loading...</div>;
   }
 
-  // Bare bones render function to hook things together
   return (
-    <div>
+    <div className={moduleStyles.miniMusicPlayer}>
       {projects.map(project => {
         return (
-          <button
-            type="button"
+          <div
+            className={moduleStyles.entry}
             key={project.id}
-            onClick={() => onPlaySong(project)}
+            onClick={() => {
+              project.id === currentProjectId
+                ? onStopSong()
+                : onPlaySong(project);
+            }}
           >
-            Play! {project.name}
-          </button>
+            <div className={moduleStyles.control}>
+              <FontAwesomeV6Icon
+                iconName={project.id === currentProjectId ? 'stop' : 'play'}
```

Comment:
Can definitely be a follow-up - but what do you think of having a loading spinner here while the code/samples are loading?

Reply:
Yes, for sure.  Will investigate in follow-up.